### PR TITLE
Chore/conversation controller functionalities

### PR DESCRIPTION
**Pull request by Yerish26**

### CODE REVIEW

File: backend/src/main/java/com/llm_service/llm_service/service/ConversationService.java [hunk 1]
```diff
@@ -11,13 +14,20 @@
 @RequiredArgsConstructor
 public class ConversationService {
     private final LLMService llmService;
+    private final ConversationPersistenceManager conversationPersistenceManager;
 
     public Conversation start() {
-        return null;
+        Conversation conversation =
```
Comment (Mgrdich) [inline backend/src/main/java/com/llm_service/llm_service/service/ConversationService.java]: maybe text should be passed by parameter, and we should rename it to `title`  in all conversation related stuff.

File: backend/src/main/java/com/llm_service/llm_service/service/ConversationService.java [hunk 2]
```diff
@@ -11,13 +14,20 @@
...
+        return conversationPersistenceManager.save(conversation);
     }
 
-    public Conversation getByID(UUID id) throws ConversationNotFoundException {
-        return null;
+    public List<Conversation> getAll() {
+        return conversationPersistenceManager.findAll();
+    }
+
+    public Optional<Conversation> getByID(UUID id) {
```
Comment (Mgrdich) [inline backend/src/main/java/com/llm_service/llm_service/service/ConversationService.java:28]: shouldn't this throw an error if the id is not found ?

Comment (Yerish26) [reply backend/src/main/java/com/llm_service/llm_service/service/ConversationService.java:28]: No, this returns optional Conversation. We check if it is empty in the controller, and then exception handler takes care of it.

File: backend/src/main/java/com/llm_service/llm_service/service/ConversationService.java (hunk 1 above)
Comment (Yerish26) [reply backend/src/main/java/com/llm_service/llm_service/service/ConversationService.java]: ?

File: backend/src/main/java/com/llm_service/llm_service/model/Conversation.java [hunk 3]
```diff
@@ -2,13 +2,13 @@
...
-import lombok.Builder;
+import lombok.Data;
 import lombok.EqualsAndHashCode;
-import lombok.Value;
+import lombok.experimental.SuperBuilder;
 
 @EqualsAndHashCode(callSuper = true)
-@Value
-@Builder(toBuilder = true)
+@SuperBuilder
```
Comment (Mgrdich) [inline backend/src/main/java/com/llm_service/llm_service/model/Conversation.java]: does this preserve to `toBuilder = true` ?

**Review by Mgrdich — APPROVED** [review]

**Pull request merged**